### PR TITLE
promisify and lipstickify

### PR DESCRIPTION
**Pull request by SamVerschueren**
pify + 💄

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,52 +1,20 @@
...
-	cb(new Error('Linux not supported'));
-}
-
-module.exports = function (cb) {
-	if (typeof cb !== 'function') {
-		throw new Error('`callback` required');
-	}
+const osx = () => runApplescript('tell app "Finder"\ntry\nPOSIX path of (insertion location as alias)\non error\nPOSIX path of (path to desktop folder as alias)\nend try\nend tell');
+const win = () => execa(path.join(__dirname, 'vendor', 'cdwhere', 'cdwhere.exe'));
+const linux = () => Promise.reject(new Error('Linux not supported'));
```
Comment (SamVerschueren) [inline index.js]: Should I just write all of these inline in the default export function?

**Comment by SamVerschueren** [comment]
Didn't want to touch the CLI to much because it needs to be extracted anyways. If you want me to revert the changes, let me know.

File: cli.js [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
+meow(`
+	Example
+	  $ current-path
+	  /Users/sindresorhus/dev/current-path
+`);
 
-currentPath(function (err, path) {
-	if (err) {
+currentPath()
+	.then(path => {
```
Comment (kevva) [inline cli.js]: `.then(console.log)`

File: test.js [hunk 3]
```diff
@@ -1,12 +1,7 @@
...
-it('should return the current file manager path', function (cb) {
-	currentPath(function (err, path) {
-		console.log('Current path:', path);
-		assert(!err, err);
-		assert(path && path.length > 0, path);
-		cb();
-	});
+test(async t => {
+	const path = await m();
+	t.true(path && path.length > 0);
```
Comment (kevva) [inline test.js]: I dunno, this is explicit and good, but `t.truthy(path)` would be the same.

File: .travis.yml [hunk 4]
```diff
@@ -1,5 +1,6 @@
 os: osx
 language: node_js
 node_js:
+  - '6'
   - '5'
```
Comment (sindresorhus) [inline .travis.yml]: Remove `5`

File: cli.js [hunk 5]
```diff
@@ -1,21 +1,19 @@
...
+		console.log(path);
+	})
+	.catch(err => {
 		console.error(err.message);
 		process.exit(1);
```
Comment (sindresorhus) [inline cli.js]: Not needed. `meow` does this.

File: index.js (hunk 1 above)
Comment (sindresorhus) [inline index.js]: 👍

File: test.js (hunk 3 above)
Comment (SamVerschueren) [inline test.js]: 👍

**Comment by SamVerschueren** [comment]
Processed feedback and added an API section in the readme. Not sure if it adds that much though.

**Comment by sindresorhus** [comment]
Thank you for all these 💄 's :)

**Comment by SamVerschueren** [comment]
💋  ;)

**Comment by SamVerschueren** [comment]
And you're more then welcome off course :)

**Pull request merged**